clippy_lints/src/functions.rs

```diff
@@ -411,6 +450,36 @@ impl<'tcx> Functions {
     }
 }
 
+fn check_result_unit_err(cx: &LateContext<'_>, decl: &hir::FnDecl<'_>, item_span: Span, fn_header_span: Span) {
+    if in_external_macro(cx.sess(), item_span) {
+        return;
+    }
+    if let hir::FnRetTy::Return(ref ty) = decl.output {
+        if let hir::TyKind::Path(ref qpath) = ty.kind {
+            if match_qpath(qpath, &crate::utils::paths::RESULT) {
+                if let Some(ref args) = last_path_segment(qpath).args {
+                    if args.args.len() == 2 {
+                        if let hir::GenericArg::Type(ref err_ty) = args.args[1] {
```

Comment:
```suggestion
                    if let [_, hir::GenericArg::Type(ref err_ty)] = args.args {
```
I think we can simplify a bit with a slice pattern